packages/webapp/src/components/Map/Modals/BulkSensorUploadModal/useValidateBulkSensorData.js

```diff
@@ -0,0 +1,219 @@
+import { useEffect, useRef, useState } from 'react';
+import * as XLSX from 'xlsx';
+import { useSelector } from 'react-redux';
+import { bulkSensorsUploadSliceSelector } from '../../../../containers/bulkSensorUploadSlice';
+
+const SENSOR_EXTERNAL_ID = 'External_ID';
+const SENSOR_NAME = 'Name';
+const SENSOR_LATITUDE = 'Latitude';
+const SENSOR_LONGITUDE = 'Longitude';
+const SENSOR_READING_TYPES = 'Reading_types';
+
+const SOIL_WATER_CONTENT = 'soil_water_content';
+const SOIL_WATER_POTENTIAL = 'soil_water_potential';
+const TEMPERATURE = 'temperature';
+
+const requiredReadingTypes = [SOIL_WATER_CONTENT, SOIL_WATER_POTENTIAL, TEMPERATURE];
+
+const requiredFields = [
+  SENSOR_EXTERNAL_ID,
+  SENSOR_NAME,
+  SENSOR_LATITUDE,
+  SENSOR_LONGITUDE,
+  SENSOR_READING_TYPES,
+];
+
+export function useValidateBulkSensorData(onUpload, t) {
+  const bulkSensorsUploadResponse = useSelector(bulkSensorsUploadSliceSelector);
+  const [disabled, setDisabled] = useState(null);
+  const [selectedFileName, setSelectedFileName] = useState('');
+  const [sheetErrors, setSheetErrors] = useState([]);
+  const [errorCount, setErrorCount] = useState(0);
+  const fileInputRef = useRef(null);
+
+  const validationFields = [
+    {
+      errorMessage: t('FARM_MAP.BULK_UPLOAD_SENSORS.VALIDATION.EXTERNAL_ID'),
+      /* eslint-disable-next-line */
```

Comment:
Please disable specific eslint rule(s) when needed, not all rules. This comment applies to several other places below.

Reply:
@anant00729

Reply:
replaced with /* eslint-disable no-useless-escape */ no-useless-escape is the only specific rule.